.github/workflows/ion-java-performance-regression-detector.yml

```diff
@@ -6,56 +6,68 @@ name: Ion Jave performance regression detector
 on:
   push:
     branches: [ master ]
+  pull_request:
+    branches: [ master ]
 
 jobs:
-  build-ion-java:
+  detetct-regression:
+    name: Detetct Regression
 
     runs-on: ubuntu-latest
 
     steps:
-      - uses: actions/checkout@v2
-
-
-
-        with:
-          fetch-depth: 2
-          submodules: recursive
       - name: Set up JDK 1.8
         uses: actions/setup-java@v1
         with:
           java-version: 1.8
-      - run: git checkout HEAD^
-      - run: mvn clean install
 
+      - name: Checkout ion-java repository from the current commit
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 2
+          submodules: recursive
+      - run: git checkout HEAD^
 
+      - name: Build ion-java from the current commit
+        run: mvn clean install
 
-      - uses: actions/checkout@v2
+      - name: Build ion-java-benchmark-cli based on the current ion-java
+        uses: actions/checkout@v2
         with:
           repository: amzn/ion-java-benchmark-cli
           ref: master
+      - run: mvn clean install
 
-      - name: Build with Maven
-        run: mvn clean install
-      - name: Verify the CLI executes
+      - name: Check the version of ion-java and ion-java-benchmark-cli
```

Comment:
A minor comment: We are not check versions of both ion-java and ion-java-benchmark-cli right? We check version of ion-java-benchmark-cli and it prints the ion-java version it is using.

Reply:
> A minor comment: We are not check versions of both ion-java and ion-java-benchmark-cli right? We check version of ion-java-benchmark-cli and it prints the ion-java version it is using.

We only need the version of ion-java, and after executing 'ion-java-benchmark-cli --version', both ion-java version and ion-java-benchmark-cli version will be printed out. Probably changing the name 'Check the version of ion-java and ion-java-benchmark-cli' to 'Check the version of ion-java' would be more reasonable.